### ISSUE TITLE
DRAGON; SYSMSG LOG missing

### ISSUE THREAD

**Issue by eswenson1** [PDP-10/its#1443]
On ES, the file DRAGON; SYSMSG LOG contains all the system log message.  This file is missing on DB.  Some daemon (on ES/PI) must be writing this file -- and for some reason, not doing so on DB.

**Comment by eswenson1** [comment]
It looks like this file is written to by dragon; rakash pfthmg (source is syseng;pft 149).  However, it only updates this file if it is found to exist.  Simply creating this file with empty contents appears to allow PFT to update it periodically.

**Comment by larsbrinkhoff** [comment]
There seems to be a difference between this file, and the messages that appear on the console.

On my NO ITS, the file only has messages about creating directories.  The console has much more, e.g. logins and writes to SYS*** files.

**Comment by eswenson1** [comment]
This is true for me on ES as well.  I don't see all messages from the console, only directory creations.

**Comment by eswenson1** [comment]
It looks like PFT only looks for LOGIN and LOGOUT dmn-push messages -- and consequently only adds these to DRAGON; SYSMSG LOG.

**Comment by eswenson1** [comment]
I thought I saw code in PFT to handle dmn-push messages of type LOGIN and LOGOUT -- and I thought it was logging those to DRAGON; SYSMSG LOG.  We should look into this further, since we clearly aren't seeing these.  Probably it is using the LOGIN and LOGOUT messages to update DRAGON; DRAGON HOARD (accounting data) and not using these notifications to update SYSMSG LOG.

**Comment by larsbrinkhoff** [comment]
Note that PEEK can display system messages.  But I only see directories and IOTLSR messages.

**Comment by eswenson1** [comment]
I assume you mean with the " command?  Yes, that appears limited too.  Probably we should investigate the SYSMSG, PEEK, PFT, and SYSJOB sources to see whether it is possible to have any one of those sinks capture all the messages.  My guess is that there isn't without ITS changes -- that some of the messages are just printed on the system console and don't go through dmn-push messages.

**Comment by larsbrinkhoff** [comment]
As a last resort, wouldn't it be possible to spy on the console?

There's also PAPSAV: #1029.

**Comment by eswenson1** [comment]
The problem with spying on the console is that unless you are always running it, you'll miss events.  You'll only see a subset of what has ever been written to the console.  What we need, I guess, is PAPSAV, which runs continuously, spys on the console, and writes the results to a file (I'm assuming that is what PAPSAV does).  Do we have code for it?  I see binaries but not source.  I guess we can reverse engineer sources from the binaries.

**Comment by eswenson1** [comment]
I grabbed CHANNA; RAKASH PAPSAV from the its-vault and tried running it on DB.  After some setup, it does a .HANG and I don't see it waking up from this call.  I forced a SYSMSG (by creating a directory), but still PAPSAV didn't wakeup from the .hang. Didn't have time to investigate on what it was hanging.

**Comment by larsbrinkhoff** [comment]
Just a binary, but it's only 300-odd words.

**Comment by larsbrinkhoff** [comment]
Maybe it's interrupt driven and was .HANGing waiting for something to happen.  The instruction before should say what would cause it to skip over the .HANG.

But this would be much easier to do with some source code to look at.  I have some other things going now that I want to finish first.

**Comment by eswenson1** [comment]
Actually, it appears to work perfectly fine. I flubbed a breakpoint after the .HANG and did a ^P.  Never got to my breakpoint and concluded that even after a SYSMSG event, it didn't wake up.  However, I checked for the file DRAGON; CDATA > and noticed that it was present and contained system messages.  I then checked my breakpoints, fixed the incorrect one, proceeded the job, and then created a new directory. My breakpoint was hit.  So it does seem to work fine. Just need to recreate the source for it!
